SwiftCompilerSources/Sources/SIL/Type.swift

```diff
@@ -113,6 +113,43 @@ public struct Type : TypeProperties, CustomStringConvertible, NoReflectionChildr
     .init(bridgedOrNil: bridged.getRawLayoutSubstitutedCountType())
   }
 
+  public var packExpansionPatternType: Type { bridged.getPackExpansionPatternType().type }
+
+  public var hasParameterPack: Bool {
```

Comment:
I think it makes sense to move (most of) these APIs to `TypeProperties` so that they are also available for `AST.Type` and `AST.CanonicalType`. The SILType implementation forwards them to the AST type, anyway